### PR TITLE
feat: add husky (#29)

### PR DESCRIPTION
**Pull request by Cahllagerfeld**
closes (#29)

### CODE REVIEW

**Comment by Cahllagerfeld** [comment]
I'm not sure if our `lint`-command does what he is intended to do: 
image

After  doing the commit my Source-Control-Tab looks like this. 

I think thats because of the `--fix`-Flag in the `lint` script.
Is this the intended behaviour?

**Comment by Cahllagerfeld** [comment]
The tests fail because, it wants to install version `0.0.1` as I bumped it `0.0.2`. So `0.0.1` isnt available anymore. Tbh I wasnt aware of this problem.. 
Shall I adjust this here as well (like in #88) or what shall I do?

**Comment by nhcarrigan-test** [comment]
Hmm, that suggests that the pre-commit hook isn't behaving properly.

Additionally, the `lint` script using the `--fix` flag effectively makes it useless in CI, as the Action will "fix" the files but they won't be committed.

My recommendation is to use a separate script for `--fix` and remove it from the `lint` that the CI uses.

**Comment by eddiejaoude** [comment]
merged #93 👍

**Comment by Cahllagerfeld** [comment]
> 
> 
> merged #93 👍

Thx!

**Comment by Cahllagerfeld** [comment]
This PR is ready for review I think

File: package.json [hunk 1]
```diff
@@ -12,8 +12,9 @@
     "start:dev": "nest start --watch",
     "start:debug": "nest start --debug --watch",
     "start:prod": "node dist/main",
-    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
+    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\"",
```
Comment (Cahllagerfeld) [inline package.json:15]: I removed the `--fix` Flag here as we are already having a `format`-Script

Comment (nhcarrigan) [reply package.json:15]: The format script runs prettier, though, which (while also helpful) won't necessarily fix `eslint` errors AFAIK.

Comment (Cahllagerfeld) [reply package.json:15]: I'd suggest a script for `lint:fix` or something similar. What do you think about that?

Comment (nhcarrigan) [reply package.json:15]: 100% agree.

**Comment by Cahllagerfeld** [comment]
After running the script I get following console: 
image
Is this a problem, or is it fine for us, that we shouldnt open issues regarding this?

**Comment by nhcarrigan** [comment]
Should be fine, I think - we could potentially look at rolling TS back to 4.1.5 if issues arise.

**Comment by Cahllagerfeld** [comment]
> 
> 
> Should be fine, I think - we could potentially look at rolling TS back to 4.1.5 if issues arise.

Okay, than I think the PR is reviewable 😄

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+npm run lint
```
Comment (eddiejaoude) [inline .husky/pre-commit:4]: Could we run the tests as well here?

```suggestion
npm run lint
npm test
npm test:e2e
```

Comment (Cahllagerfeld) [reply .husky/pre-commit:4]: Yes, we can do this. But then we probably cant push any WIP changes (like step-definitions) for example to the PR. They have to be resolved locally then, or am I wrong?

Comment (nhcarrigan-test) [reply .husky/pre-commit:4]: Ordinarily yes, though you can pass the `--no-verify` flag to skip the pre-commit hooks.

Comment (nhcarrigan) [reply .husky/pre-commit:4]: Ordinarily yes, though you can pass the --no-verify flag to skip the pre-commit hooks.

Comment (Cahllagerfeld) [reply .husky/pre-commit:4]: Added the scripts to the hook

**Pull request merged**